lightning/src/ln/channelmanager.rs

```diff
@@ -3760,6 +3812,7 @@ where
 												payment_hash,
 												purpose: purpose(),
 												amount_msat,
+												counterparty_skimmed_fee_msat: total_value.saturating_sub(amount_msat),
```

Comment:
In the case where a counterparty overshoots the amount in the onion, I think this would report the skimmed fee inaccurately? E.g. the `counterparty_skimmed_fee_msat` would be offset by `amount_msat - total_value`. It probably wouldn't happen very often and the offset probably wouldn't be much, but figured I'd still make note of it.

Reply:
I think it'd be possible by adding the skimmed fee to `PendingHTLCRouting::Receive`/`ReceiveKeysend` (which seems possible since the skimmed fee gets sent through `construct_recv_pending_htlc_info`), and then probably keeping this on `ClaimableHTLC` and using those when finally calculating? Not sure if it's worth trying to fit it in here, could maybe be followup

Reply:
Nice catch! Fixed. I didn't want to just trust whatever the counterparty set as the skimmed fee since they could technically lie about that, so it's now a mix of the previous and new solution, PTAL.